src/plugins/data/public/search/session/session_service.ts

```diff
@@ -251,7 +260,15 @@ export class SessionService {
    */
   public getSearchOptions(
     sessionId: string
-  ): Required<Pick<ISearchOptions, 'sessionId' | 'isRestore' | 'isStored'>> {
+  ):
+    | Required<Pick<ISearchOptions, 'sessionId' | 'isRestore' | 'isStored'>>
+    | { sessionId: undefined } {
+    // in case user doesn't have permissions to search session, do not forward sessionId to the server
+    // because user most likely also doesn't have access to `background-session` SO
+    if (!this.searchSessionsCapabilities?.create) {
+      return { sessionId: undefined };
```

Comment:
Noticed that after this pr https://github.com/elastic/kibana/pull/89570 we track all searches in es and `trackId` would through in case a user doesn't have access to `background-session` SO. 
To avoid this ensure we don't send `sessionId` to the server in case a user doesn't have access to work with sessions